fix libpng sdl error

Sirox0 opened the pull request:
when one of my testers with armv7 tried to launch the app, this happened
![Screenshot_20221029_152149_com shadowmario psychengine](https://user-images.githubusercontent.com/71220271/198851443-61d6136e-508c-4f51-ae2f-9972e843f769.jpg)
i searched a bit and found that neon thing in png applies only for arm64 and working only arm64, i changed to all arches and it worked, as i understand it, that error appears on all ndks.

Sirox0 committed:
Update png-files.xml

player-03 commented:
There is [only one reference](https://sourceforge.net/p/libpng/code/ci/a40189cf881e9f0db80511c382292a5604c3c3d1/tree/pngpriv.h#l2133) to that function in all of Lime and libpng, and that only happens if `PNG_ARM_NEON_IMPLEMENTATION == 1`.

libpng explains `PNG_ARM_NEON_IMPLEMENTATION` [starting here](https://sourceforge.net/p/libpng/code/ci/a40189cf881e9f0db80511c382292a5604c3c3d1/tree/pngpriv.h#l143). I had assumed that [line 176](https://sourceforge.net/p/libpng/code/ci/a40189cf881e9f0db80511c382292a5604c3c3d1/tree/pngpriv.h#l176) was the deciding factor, but it's also possible that `__ARM_NEON__` is defined, in which case it'll use implementation `1` by default. I don't think png-files.xml can test for `__ARM_NEON__`, so maybe we have to compile the files regardless.

Edit: [they also mention that `__ARM_NEON__` can be set automatically](https://sourceforge.net/p/libpng/code/ci/a40189cf881e9f0db80511c382292a5604c3c3d1/tree/pngpriv.h#l110). That's probably what happened.

player-03 committed:
Only compile Arm-specific files for Arm devices.

These files are only required if `PNG_ARM_NEON_IMPLEMENTATION == 1` (the default). Since we can't test it here, we just have to assume it's the case on all Arm devices. (But only Arm devices.)

player-03 commented:
The only other thing is we don't need to compile Arm-specific code for x86 devices. `__ARM_NEON__` won't be set in that case.
